### PR TITLE
update with IPv6+IPv4 family addition to OSM/MC

### PR DESCRIPTION
**Pull request by PratikDeoghare**
**What this PR does / why we need it**:
OSM/MC got new IP family IPv6+IPv4. This PR updates KKP to work with this new OSM/MC.  
Ensures existing dual-stack tests still pass.

**Which issue(s) this PR fixes**:

Towards partial fulfillment of https://github.com/kubermatic/kubeone/issues/2492

**What type of PR is this?**

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change? Then add your Release Note here**:

```release-note
NONE
```

**Documentation**:

```documentation
NONE
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -340,3 +338,11 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.13.9 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
```
Comment (PratikDeoghare) [inline go.mod]: Remove after https://github.com/kubermatic/operating-system-manager/pull/242 merges

**Review by embik — APPROVED** [review]
/approve

**Pull request merged**